### PR TITLE
CHE-3199: add cpu limits configuration

### PR DESCRIPTION
**Pull request by garagatyi**
### What does this PR do?
Adds limits for containers of a workspace.

I added next properties to configure limits:
**che.docker.parent_cgroup=NULL**
Sets parent cgroup for cgroups of containers created by workspaces.
This allows an admin to set custom cgroup limitations to all containers of workspaces by configuring cgroups.
Example:
/my_group
my_another_group
**che.docker.cpuset_cpus=NULL**
Sets set of CPUs that can be used by each container of started workspace.
Example:
0-3
1,4
**che.docker.cpu_period=0
che.docker.cpu_quota=0**
These 2 properties set limits on CPU consumption by containers of started workspaces.
Period sets amount of units per CPU core.
Quota sets amount of units available for container per whole CPU.
Max value of quota could be period * number of CPU cores in a system.
Example:
che.docker.cpu_period=5000
che.docker.cpu_quota=10000

### What issues does this PR fix or reference?
#3199

### CODE REVIEW

**Comment by garagatyi** [comment]
@skabashnyuk @sleshchenko @mmorhun Please review

**Comment by TylerJewell** [comment]
Needs to have additions added into the che.env file template (with descriptions for users to know how to configure) and in the /docs/admin.md file to explain when / how to configure this.

Does this apply to codenvy at all?

**Comment by TylerJewell** [comment]
I spent about 30 minutes reviewing this tonight.  We need to make some modifications first, before we commit this:
1. We need the appropriate parameters added to the template che.env file at
https://github.com/eclipse/che/blob/master/dockerfiles/init/manifests/che.env

2. I'm worried that the average (or even the above average) admin / user will not understand what a cgropu, a cpushare, cpu-period will be. So just copy / pasting the explanation from the Docker web site is not good enough, as that explanation is not clear. So in the descriptions for the parameters that are inserted into che.env, try to provide explanations on the meaning for each term assuming that the admin does not know what those terms mean or understands that it comes from Docker.  It's ok to include links to pages on Docker's site that explain those terms in addition to our explanation.

3. Finally, in the che.env file, we should probably place a sample configuration that informs the admin how to configure Che to use 50% of one cpu, in a 2 cpu system. If we can just expose that basic scenario and how you would set these parameters, that would be helpful.

**Comment by garagatyi** [comment]
@TylerJewell about # 2. It is my own explanation - not the one from docker docs. I'll try to rewrite docs in a way you want. Also feel free to provide your own variant here or in a private chat. 
Also I do believe that if admin doesn't know what is cgroup he must not use that property because it requires manual configuring of each docker node to work properly. So average admin that doesn't know how to do that should read about that and how to set it in his own distributive (because it reaaly depends on distributive of linux).

File: plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/params/BuildImageParams.java [hunk 1]
```diff
@@ -426,6 +483,9 @@ public String toString() {
                ", noCache=" + noCache +
                ", removeIntermediateContainer=" + removeIntermediateContainer +
                ", forceRemoveIntermediateContainers=" + forceRemoveIntermediateContainers +
+               ", cpusetCpus=" + cpusetCpus +
```
[the remark this answers is not in the document]
Comment (garagatyi) [reply plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/params/BuildImageParams.java]: Good catch!

File: plugins/plugin-docker/che-plugin-docker-client/src/test/java/org/eclipse/che/plugin/docker/client/params/BuildImageParamsTest.java [hunk 2]
```diff
@@ -211,205 +231,63 @@ public void shouldThrowIllegalStateExceptionIfAddFileAfterSetRemote() {
 
     @Test
     public void repositoryParameterShouldEqualsNullIfItNotSet() {
-        buildImageParams.withTag(TAG)
```
[the remark this answers is not in the document]
Comment (garagatyi) [reply plugins/plugin-docker/che-plugin-docker-client/src/test/java/org/eclipse/che/plugin/docker/client/params/BuildImageParamsTest.java:214]: Then it is possible that second setting breaks everything or third or ... Should we test all these cases? I think no, then it doesn't make sense to set all possible fields in each test.

**Review by mmorhun — APPROVED** [review]
Technical side looks good to me.

**Comment by garagatyi** [comment]
@TylerJewell I updated docs, please review

**Comment by TylerJewell** [comment]
I cleaned up the che.env configuration options.  Ok to merge.

**Review by skabashnyuk — APPROVED** [review]

**Comment by garagatyi** [comment]
@TylerJewell I've finished with documentation

**Pull request merged**